src/services/user/setWPUUser.js

```diff
@@ -0,0 +1,30 @@
+const {successResponse, errorResponse} = require('../../utils');
+const {adminBlockUserProcess} = require('../../processes/scoring/admin-block-unblock-user');
+const UsersFunction = require('../../databases/functions/users');
+
+const setWPUUser = async (req, res) => {
+  try {
+    const {userId, score} = req.body;
+    if (!userId) {
+      return errorResponse(res, 'userId is required', 400);
+    }
+    if (!score) {
+      return errorResponse(res, 'score is required', 400);
+    }
+
+    try {
+      await UsersFunction.setWPUUser(userId, score);
+    } catch (error) {
+      return errorResponse(res, 'Failed to update w pu user', 500);
+    }
+    const updateResult = await adminBlockUserProcess(userId);
+    return successResponse(res, 'update w pu user was successful', updateResult);
+  } catch (error) {
+    console.log(error);
+    return errorResponse(res, error.toString(), 500);
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in this function could be improved. Currently, if an error occurs during the `UsersFunction.setWPUUser(userId, score)` operation, it is caught and a response is returned, but the function continues to execute `adminBlockUserProcess(userId)`. This could lead to unexpected behavior or further errors. Consider adding a `return` statement in the catch block after line 18 to prevent further execution when an error occurs.

Also, logging the error on line 23 with `console.log(error)` is not a good practice for production code. It would be better to use a proper logging library that can provide more context about the error and handle different log levels.

```diff
    try {
      await UsersFunction.setWPUUser(userId, score);
    } catch (error) {
      return errorResponse(res, 'Failed to update w pu user', 500);
+       return;
    }
    const updateResult = await adminBlockUserProcess(userId);
    return successResponse(res, 'update w pu user was successful', updateResult);
  } catch (error) {
-      console.log(error);
+      logger.error(error);
    return errorResponse(res, error.toString(), 500);
  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->